src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionAuthorizationRequestedQueueConsumer.kt

```diff
@@ -0,0 +1,112 @@
+package it.pagopa.ecommerce.eventdispatcher.queues
+
+import com.azure.core.util.BinaryData
+import com.azure.core.util.serializer.TypeReference
+import com.azure.spring.messaging.AzureHeaders
+import com.azure.spring.messaging.checkpoint.Checkpointer
+import it.pagopa.ecommerce.commons.documents.BaseTransactionEvent
+import it.pagopa.ecommerce.commons.documents.v1.TransactionAuthorizationRequestedEvent as TransactionAuthorizationRequestedEventV1
+import it.pagopa.ecommerce.commons.documents.v2.TransactionAuthorizationRequestedEvent as TransactionAuthorizationRequestedEventV2
+import it.pagopa.ecommerce.commons.queues.QueueEvent
+import it.pagopa.ecommerce.commons.queues.StrictJsonSerializerProvider
+import it.pagopa.ecommerce.commons.queues.TracingInfo
+import it.pagopa.ecommerce.eventdispatcher.exceptions.InvalidEventException
+import it.pagopa.ecommerce.eventdispatcher.utils.DeadLetterTracedQueueAsyncClient
+import java.util.*
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.integration.annotation.ServiceActivator
+import org.springframework.messaging.handler.annotation.Header
+import org.springframework.messaging.handler.annotation.Payload
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+
+/**
+ * Event consumer for transactions to refund. These events are input in the event queue only when a
+ * transaction is stuck in an REFUND_REQUESTED state **and** needs to be reverted
+ */
+@Service("TransactionAuthorizationRequestedQueueConsumer")
+class TransactionAuthorizationRequestedQueueConsumer(
+  @Autowired
+  private val queueConsumerV2:
+    it.pagopa.ecommerce.eventdispatcher.queues.v2.TransactionAuthorizationRequestedQueueConsumer,
+  @Autowired private val deadLetterTracedQueueAsyncClient: DeadLetterTracedQueueAsyncClient,
+  @Autowired private val strictSerializerProviderV1: StrictJsonSerializerProvider,
+  @Autowired private val strictSerializerProviderV2: StrictJsonSerializerProvider
+) {
+
+  var logger: Logger = LoggerFactory.getLogger(TransactionsRefundQueueConsumer::class.java)
+
+  fun parseEvent(payload: ByteArray): Mono<Pair<BaseTransactionEvent<*>, TracingInfo?>> {
+    val data = BinaryData.fromBytes(payload)
+    val jsonSerializerV1 = strictSerializerProviderV1.createInstance()
+    val jsonSerializerV2 = strictSerializerProviderV2.createInstance()
+    val transactionAuthorizationRequestedEventV1 =
+      data
+        .toObjectAsync(
+          object : TypeReference<QueueEvent<TransactionAuthorizationRequestedEventV1>>() {},
+          jsonSerializerV1)
+        .map { it.event to it.tracingInfo }
+        .onErrorResume {
+          logger.debug(ERROR_PARSING_EVENT_ERROR, it)
+          Mono.empty()
+        }
```

Comment:
this logic has to be implemented only for v2 transactions service handlers, right? so here we can exclude to receive a v1 event